### PR TITLE
1/3 Port PDH helpers from Core

### PR DESCRIPTION
**Pull request by james-bebbington**
**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/1088

**Description:** As a precursor to creating a Windows perf counters receiver component, this PR:

- Ports package `pdh` from Core (was removed from Core in https://github.com/open-telemetry/opentelemetry-collector/pull/1870)
- Ports third_party package `win_perf_counters` from Core (was removed from Core in https://github.com/open-telemetry/opentelemetry-collector/pull/1870)

**See also:** https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1211

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@james-bebbington would it be possible to break into 2 PRs: 
1. Copy old (deleted) code as is. We know it was already reviewed so we can merge this without thorough review.
2. Any changes you need to make. This one we can review more carefully.
Would that work?

**Comment by james-bebbington** [comment]
> @james-bebbington would it be possible to break into 2 PRs:
> 
> 1. Copy old (deleted) code as is. We know it was already reviewed so we can merge this without thorough review.
> 2. Any changes you need to make. This one we can review more carefully.
>    Would that work?

Alright done. This PR is (1) & https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1211 is (2)

**Comment by tigrannajaryan** [comment]
> Alright done. I updated this PR & the description so this is now (1) & #1211 is (2)

Thank you.

File: receiver/windowsperfcountersreceiver/internal/third_party/telegraf/win_perf_counters/LICENSE [hunk 1]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
+
+Copyright (c) 2015-2019 InfluxData Inc.
```
Comment (asuresh4) [inline receiver/windowsperfcountersreceiver/internal/third_party/telegraf/win_perf_counters/LICENSE:3]: 2020?

File: receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build windows
+
+package pdh
```
Comment (asuresh4) [inline receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go:17]: Do we currently run tests on Windows for this repo?

**Review by asuresh4 — APPROVED** [review]

File: receiver/windowsperfcountersreceiver/internal/third_party/telegraf/win_perf_counters/LICENSE (hunk 1 above)
Comment (james-bebbington) [reply receiver/windowsperfcountersreceiver/internal/third_party/telegraf/win_perf_counters/LICENSE:3]: This was the LICENSE file brought over from Telegraf. I don't think I should change it should I?

File: receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go (hunk 2 above)
Comment (james-bebbington) [reply receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go:17]: Good question - it doesn't look like we do. This code is copied from Core so I can promise it succeeds 😄 

I'll create a followup PR to add a **windows-build** step for Contrib, similar to what @bogdandrutu previously added for Core.

File: receiver/windowsperfcountersreceiver/internal/third_party/telegraf/win_perf_counters/LICENSE (hunk 1 above)
Comment (asuresh4) [reply receiver/windowsperfcountersreceiver/internal/third_party/telegraf/win_perf_counters/LICENSE:3]: I believe it was updated. https://github.com/influxdata/telegraf/blob/master/LICENSE#L3.

File: receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go (hunk 2 above)
Comment (asuresh4) [reply receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go:17]: 👍 SGTM.

Comment (james-bebbington) [reply receiver/windowsperfcountersreceiver/internal/pdh/pdh_test.go:17]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/1228

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**